### PR TITLE
💻 #2271 Clinical Data download

### PR DESCRIPTION
**Pull request by demariadaniel**
# Description of changes

* Download buttons for All Clinical Data + Currently Selected Clinical Data

Zenhub: https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/icgc-argo/platform-ui/2271
Api: https://github.com/icgc-argo/platform-api/pull/606
Service: https://github.com/icgc-argo/argo-clinical/pull/819

## Type of Change

- [ ] Bug
- [ ] Dependency updates
- [x] Feature
- [ ] Refactoring
- [ ] Release candidate
- [ ] Styling
- [ ] Testing
- [ ] Other (please describe)

## Checklist before requesting review

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- Back-end (select one):
  - [ ] Back-end PRs merged, tested on develop, and linked in this PR
  - [ ] No back-end changes
- [x] Manually tested changes
- [x] Added copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: components/pages/submission-system/program-submitted-data/DownloadButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    const { GATEWAY_API_ROOT } = getConfig();
+
+    const url = urlJoin(
+      GATEWAY_API_ROOT,
+      `/clinical/program/${programShortName}/clinical-data-tsv?${query}`,
+    );
+
+    setButtonLoadingState({ ...buttonLoadingState, isDownloadAllLoading: true });
+
+    fetch(url, {
```
Comment (samrichca) [inline components/pages/submission-system/program-submitted-data/DownloadButtons.tsx:123]: we have a fetch function & download function in `useAuthContext` - would those work here?

File: components/pages/submission-system/program-submitted-data/DownloadButtons.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  searchResults: ClinicalSearchResults[];
+  entityTypes: string[];
+  completionState: CompletionStates;
+}) => {
+  const toaster = useToaster();
+
+  const { shortName: programShortName } = usePageQuery<{ shortName: string }>();
+
+  const [buttonLoadingState, setButtonLoadingState] = React.useState({
+    isDownloadAllLoading: false,
```
Comment (samrichca) [inline components/pages/submission-system/program-submitted-data/DownloadButtons.tsx:83]: is it possible for this object to have other properties, or just this one?

File: components/pages/submission-system/program-submitted-data/index.tsx [hunk 3]
```diff
@@ -233,28 +232,12 @@ export default function ProgramSubmittedData() {
...
-                    css={css`
-                      padding-right: 4px;
-                    `}
-                    name="download"
-                    fill={noData ? 'grey_1' : 'accent2_dark'}
-                    height="12px"
-                  />
-                  {clinicalEntityDisplayNames[currentEntity]} Data
-                </Button>
+                <ClinicalDownloadButton
```
Comment (samrichca) [inline components/pages/submission-system/program-submitted-data/index.tsx:235]: good code reuse 👍

File: components/pages/submission-system/program-submitted-data/DownloadButtons.tsx (hunk 2 above)
Comment (demariadaniel) [reply components/pages/submission-system/program-submitted-data/DownloadButtons.tsx:83]: Jus this one, copy/paste from other button component, could def be simplified to just a boolean

File: components/pages/submission-system/program-submitted-data/DownloadButtons.tsx (hunk 1 above)
Comment (demariadaniel) [reply components/pages/submission-system/program-submitted-data/DownloadButtons.tsx:126]: Possibly but they handle file names differently. I'll try it out in the morning.

Comment (demariadaniel) [reply components/pages/submission-system/program-submitted-data/DownloadButtons.tsx:126]: OK so the `downloadFileWithEgoToken` function works great _except_ it's replacing our nice file name w/ a hashed value and I'm not entirely sure how or why. Looking into it.

Comment (demariadaniel) [reply components/pages/submission-system/program-submitted-data/DownloadButtons.tsx]: This is now working with a tiny change necessary for the clinical service to get the filenames working properly
https://github.com/icgc-argo/argo-clinical/pull/822

**Review by samrichca — APPROVED** [review]
thanks for making those changes, great work! 🚀

**Pull request merged**